From edce37fa32f7704fc25d3576f40f0bd7d0fcd7de Mon Sep 17 00:00:00 2001
From: althafvly <althafvly@gmail.com>
Date: Sat, 30 Apr 2022 16:07:03 +0530
Subject: [PATCH] fonts: Dynamically add additional fonts

eg: ADDITIONAL_FONTS_FILE := $(LOCAL_PATH)/google-fonts/fonts_customization.xml
Change-Id: Ibf12c69339f2ced37c1e9181cc38ebb5c7ee3857
---

diff --git a/fonts/etc/Android.mk b/fonts/etc/Android.mk
index 64bc664..46c5bc6 100644
--- a/fonts/etc/Android.mk
+++ b/fonts/etc/Android.mk
@@ -2,8 +2,24 @@
 
 include $(CLEAR_VARS)
 LOCAL_MODULE := fonts_customization.xml
-LOCAL_SRC_FILES := fonts_customization.xml
-LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE_CLASS := ETC
+
+DEFAULT_FONTS_FILE := packages/overlays/Lineage/fonts/etc/fonts_customization.xml
+
+ifdef ADDITIONAL_FONTS_FILE
+ADDITIONAL_FONTS_SCRIPT := vendor/lineage/tools/fonts_combiner.py
+FINAL_FONTS_FILE := $(local-generated-sources-dir)/fonts_customization.xml
+
+$(FINAL_FONTS_FILE): PRIVATE_SCRIPT := $(ADDITIONAL_FONTS_SCRIPT)
+$(FINAL_FONTS_FILE): PRIVATE_ADDITIONAL_FONT_FILE := $(ADDITIONAL_FONTS_FILE)
+$(FINAL_FONTS_FILE): $(ADDITIONAL_FONTS_SCRIPT) $(DEFAULT_FONTS_FILE)
+	rm -f $@
+	python3 $(PRIVATE_SCRIPT) $(DEFAULT_FONTS_FILE) \
+	$(PRIVATE_ADDITIONAL_FONT_FILE) $(FINAL_FONTS_FILE)
+else
+FINAL_FONTS_FILE := $(DEFAULT_FONTS_FILE)
+endif
+
+LOCAL_PREBUILT_MODULE_FILE := $(FINAL_FONTS_FILE)
 LOCAL_PRODUCT_MODULE := true
 include $(BUILD_PREBUILT)
